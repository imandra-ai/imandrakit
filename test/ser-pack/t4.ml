type t1 =
  | F of (string[@as_bytes])
  | G of string
[@@deriving show, serpack]

let x = Imandrakit_ser_pack.to_cbor_string t1_to_serpack (F "hello\x000world");;

Format.printf "x: %S@." x;;
Format.printf "len: %d@." (String.length x)

let t1' = Imandrakit_ser_pack.of_cbor_string_exn t1_of_serpack x;;

Format.printf "deser: %a@." pp_t1 t1'
