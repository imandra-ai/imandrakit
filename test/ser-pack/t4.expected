x: "\162ak\198\000ah\129\130\000lhello\0000world"
len: 23
deser: (T4.F "hello\0000world")
